### Analysis
#### Objective
Determine if the new machine packs faster than the old machine using a one-sided t-test.

#### Hypotheses
- *Null Hypothesis* (H₀): No difference in packing times between new and old machine.
- *Alternative Hypothesis* (H₁): New machine packs faster than the old machine.
Test Statistics

#### Test Statistics


In [1]:
# Importing necessary libraries
import numpy as np
from scipy import stats

# Machine time data
new_machine = np.array([42.1, 41, 41.3, 41.8, 42.4, 42.8, 43.2, 42.3, 41.8, 42.7])
old_machine = np.array([42.7, 43.6, 43.8, 43.3, 42.5, 43.5, 43.1, 41.7, 44, 44.1])

# Student GPA data
sophomores = np.array([3.04, 1.71, 3.3, 2.88, 2.11, 2.6, 2.92, 3.6, 2.28, 2.82, 3.03, 3.13, 2.86, 3.49, 3.11, 2.13, 3.27])
juniors = np.array([2.56, 2.77, 2.7, 3, 2.98, 3.47, 3.26, 3.2, 3.19, 2.65, 3, 3.39, 2.58])

# Calculations for the machine packing times
# Assuming equal variances for a pooled two-sample t-test
mean_new = np.mean(new_machine)
mean_old = np.mean(old_machine)
std_new = np.std(new_machine, ddof=1)
std_old = np.std(old_machine, ddof=1)
n_new = len(new_machine)
n_old = len(old_machine)

sp = np.sqrt(((n_new - 1) * std_new**2 + (n_old - 1) * std_old**2) / (n_new + n_old - 2))
t_stat_machine = (mean_new - mean_old) / (sp * np.sqrt(1/n_new + 1/n_old))
df_machine = n_new + n_old - 2

# Calculations for the student GPAs
mean_soph = np.mean(sophomores)
mean_jun = np.mean(juniors)
std_soph = np.std(sophomores, ddof=1)
std_jun = np.std(juniors, ddof=1)
n_soph = len(sophomores)
n_jun = len(juniors)

s = np.sqrt(std_soph**2 / n_soph + std_jun**2 / n_jun)
t_stat_gpa = (mean_soph - mean_jun) / s
df_gpa = ((std_soph**2 / n_soph + std_jun**2 / n_jun)**2) / \
         (((std_soph**2 / n_soph)**2 / (n_soph - 1)) + ((std_jun**2 / n_jun)**2 / (n_jun - 1)))

t_stat_machine, df_machine, t_stat_gpa, df_gpa

(-3.3972307061176026, 18, -0.9231495630900276, 26.631591966307603)

The test statistic calculated is -3.397 with degrees of freedom = 18

This suggests the new machine might not be faster, and potentially slower than the old one

-------------------------------


### Analysis of Student GPAs
####  Objective: 
- To verify if there's a significant difference in GPAs between sophomores and juniors without assuming equal variances.

#### Hypotheses
- Null Hypothesis (H₀): Mean GPAs of sophomores and juniors are equal.
- Alternative Hypothesis (H₁): Mean GPAs of sophomores and juniors differ.
#### Test Statistics
The test statistic calculated is -0.923 with degrees of freedom ≈ 26.63

The results indicate insufficient evidence to conclude a difference in mean GPAs